appscale/tools/agents/azure_agent.py

```diff
@@ -265,31 +266,77 @@ def describe_instances(self, parameters, pending=False):
     private_ips = []
     instance_ids = []
 
-    public_ip_addresses = network_client.public_ip_addresses.list(resource_group)
-    for public_ip in public_ip_addresses:
-      public_ips.append(public_ip.ip_address)
+    public_ip_addresses, error = self.run_and_return_exception(
```

Comment:
Is this the case whether we could possibly run into an exception while trying to just list the IP addresses in the resource group? The one you said @whoarethebritons you ran into while testing? And this is basically us being defensive right or did we always run into it?

Reply:
@tmarballi yes exactly I ran into it in testing, one of the reasons is because describe instances is now a part of run_instances in the Infrastructure Manager so we need to be able to handle a "too many requests" issue.